### PR TITLE
Add PHP7.4 support to postPackage

### LINKED ISSUE #28: PHP 7.4 support
Hi, setting up Drupal 7 via composer with PHP 7.4 currently fails with the following error:

```shell
$ composer -v create-project drupal-composer/drupal-project:7.x-dev some-dir --no-interaction
Installing drupal-composer/drupal-project (7.x-dev 267d7fe14c74af8c04fc84c7f483434eca35ef90)
  - Installing drupal-composer/drupal-project (7.x-dev 267d7fe): Cloning 267d7fe14c74af8c04fc84c7f483434eca35ef90 from cache
Created project in some-dir
> pre-update-cmd: DrupalProject\composer\ScriptHandler::checkComposerVersion
Loading composer repositories with package information
Updating dependencies (including require-dev)
Dependency resolution completed in 0.255 seconds
Analyzed 6353 packages to resolve dependencies
Analyzed 403588 rules to resolve dependencies
Package operations: 29 installs, 0 updates, 0 removals
Installs: composer/installers:v1.7.0, cweagans/composer-patches:1.6.7, drupal-composer/preserve-paths:0.1.5, composer/semver:1.5.0, drupal/drupal:7.67.0, drupal/composer_autoloader:1.3.0, symfony/polyfill-ctype:v1.13.1, webmozart/assert:1.6.0, webmozart/path-util:2.3.0, webflo/drupal-finder:1.2.0, pear/console_table:v1.3.1, symfony/finder:v3.4.36, symfony/event-dispatcher:v3.4.36, psr/log:1.1.2, symfony/debug:v4.4.1, symfony/polyfill-mbstring:v1.13.1, symfony/console:v3.4.36, symfony/var-dumper:v3.4.36, symfony/yaml:v3.4.36, dflydev/dot-access-data:v1.1.0, consolidation/output-formatters:3.5.0, consolidation/annotated-command:2.12.0, jakub-onderka/php-console-color:v0.2, jakub-onderka/php-console-highlighter:v0.4, dnoegel/php-xdg-base-dir:0.1, nikic/php-parser:v4.3.0, psy/psysh:v0.9.11, drush/drush:8.3.2, symfony/filesystem:v3.4.36
  - Installing composer/installers (v1.7.0): Downloading (100%)
 Extracting archive  - Installing cweagans/composer-patches (1.6.7): Downloading (100%)
 Extracting archiveNo patches found for cweagans/composer-patches.
Gathering patches for root package.
Gathering patches for dependencies. This might take a minute.
Found 1 patches for cweagans/composer-patches.
  - Installing drupal-composer/preserve-paths (0.1.5): Downloading (100%)
 Extracting archiveNo patches found for drupal-composer/preserve-paths.


  [ErrorException]
  Trying to access array offset on value of type null


Exception trace:
 () at /var/www/html/some-dir/vendor/drupal-composer/preserve-paths/src/PluginWrapper.php:88
 Composer\Util\ErrorHandler::handle() at /var/www/html/some-dir/vendor/drupal-composer/preserve-paths/src/PluginWrapper.php:88
 DrupalComposer\PreservePaths\PluginWrapper->postPackage() at /var/www/html/some-dir/vendor/drupal-composer/preserve-paths/src/Plugin.php:69
 DrupalComposer\PreservePaths\Plugin->postPackage() at n/a:n/a
 call_user_func() at phar:///usr/local/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:176
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///usr/local/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:116
 Composer\EventDispatcher\EventDispatcher->dispatchPackageEvent() at phar:///usr/local/bin/composer/src/Composer/Installer.php:622
 Composer\Installer->doInstall() at phar:///usr/local/bin/composer/src/Composer/Installer.php:229
 Composer\Installer->run() at phar:///usr/local/bin/composer/src/Composer/Command/CreateProjectCommand.php:195
 Composer\Command\CreateProjectCommand->installProject() at phar:///usr/local/bin/composer/src/Composer/Command/CreateProjectCommand.php:145
 Composer\Command\CreateProjectCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:245
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:835
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:267
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:106
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:61
 require() at /usr/local/bin/composer:24

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-reqs] [--] [<package>] [<directory>] [<version>]

```

### PR DESCRIPTION
**Pull request by Arkener**
Fixes #28

### CODE REVIEW

**Comment by jackthomasatl** [comment]
Can this please be merged / built?

**Comment by leymannx** [comment]
@Arkener – Can you request a review?

File: src/PluginWrapper.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function postPackage(PackageEvent $event)
     {
         $packages = $this->getPackagesFromEvent($event);
         $key = $this->getUniqueNameFromPackages($packages);
-        if ($this->preservers[$key]) {
+        if (!empty($this->preservers[$key])) {
```
Comment (leymannx) [inline src/PluginWrapper.php]: Shouldn't it rather be `if ($this->preservers[$key] && !empty($this->preservers[$key]))`?

Comment (jackthomasatl) [reply src/PluginWrapper.php]: I don't think that would fix the issue. !empty first will both verify the key is present and validate it contains something. The original patch fixes the issue on my end.

Comment (jackthomasatl) [reply src/PluginWrapper.php]: Perhaps `!empty($this->preservers) && $this->preservers[$key]` might make better sense.

Comment (jackthomasatl) [reply src/PluginWrapper.php]: When debugging, I'm seeing preserve-paths get called to validate itself. But ->preservers is null at that point.

Comment (leymannx) [reply src/PluginWrapper.php]: Ah yeah! You are right. Just read the docs again.

> `empty()` is essentially the concise equivalent to `!isset($var) || $var == false`

So the original patch should be fine indeed.

Comment (jackthomasatl) [reply src/PluginWrapper.php]: With the exception that the problem here is ->preservers is null when the defect is triggered. Its not that the array has null values, but the the array does not exist yet.

Comment (Arkener) [reply src/PluginWrapper.php]: `empty()` will not trigger an error if the variable doesn't exist, this means that `!empty($this->preservers[$key])` will not throw a warning, even if `$this->preservers` is null.

Comment (jackthomasatl) [reply src/PluginWrapper.php]: Agreed, I've confirmed this PR resolves the issue. I sent an email to Johannes Haseitl to get his attention to the PR. Can anyone else try to reach out and get some traction on this?

Comment (claylibrarymarket) [reply src/PluginWrapper.php:88]: I've also tried to get into contact with @derhasi on the Drupal Slack server to no avail.

Can we push this through without his approval? This change doesn't seem dangerous.

Comment (derhasi) [reply src/PluginWrapper.php:88]: I‘m sorry I did not react to this. I am currently not really maintaining any opensource projects 😞.

This change looks pretty straightforward. Thanks for your fix @Arkener

**Comment by EnnioWolsink** [comment]
@derhasi @plebianmvi Can you tag these changes in a new release also? Composer still can't download this fix as the latest tag is still 2 years old (0.1.5).

**Comment by jackthomasatl** [comment]
Does anyone know how to get composer to tag a package as abandoned?

**Comment by leymannx** [comment]
@jackthomasatl – I think only the user who created the package can mark it as abandoned. See https://stackoverflow.com/q/40946151/2199525. Maybe open an issue "Mark drupal-composer/preserve-paths as abandoned" here.

**Pull request merged**